### PR TITLE
MNT:Add path to git sympy in PYTHONPATH environment variable

### LINKED ISSUE #17369: test and doctest scripts do not work from other directories
The `path_hack` in the `test` and `doctest` scripts is supposed to make it so that something like

```
cd bin
./doctest
```

still tests the git SymPy. 

But the problem is that the tests and doctests run in a subprocess by default, which does not inherit the sys.path. So the above will test the globally installed SymPy unless `--no-subprocess` is used. 

The simplest solution would be for `path_hack` to set PYTHONPATH in os.environ so that it gets inherited. Or maybe it should be done in the run_tests_in_subprocess function, I'm not sure.

### PR DESCRIPTION
**Pull request by sameshl**
add the relative path to the git Sympy with `sys.path.insert` to PYTHONPATH

closes #17369

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
I don't think this works. The sys.path isn't inherited by the subprocess. Actually the current sympy should already be the first on the sys.path.

File: sympy/utilities/runtests.py [hunk 1]
```diff
@@ -225,6 +225,9 @@ def run_in_subprocess_with_hash_randomization(
 
     function_kwargs = function_kwargs or {}
 
+    # Add path to the git sympy directory to PYTHONPATH envirnoment variable
+    # before importing sympy
+    sys.path.insert(0, "../../.")
```
Comment (asmeurer) [inline sympy/utilities/runtests.py]: ../../. is relative to the user's current directory, not the location of this file. You need to use `__file__` to get the directory of this file.

**Comment by oscarbenjamin** [comment]
I would prefer not to mess with PYTHONPATH since that should be for users to control. You could set the current working directory (cwd) of the child process in the subprocess call though:
https://docs.python.org/3/library/subprocess.html#popen-constructor

**Comment by asmeurer** [comment]
That doesn't guarantee it will import the right sympy. 

But maybe you are saying if a user sets PYTHONPATH that should override this. I guess that seems valid, so maybe setting cwd is more correct.

**Pull request closed without merging**